api/v1/recorder/recorder.proto

```diff
@@ -49,6 +49,26 @@ message StartRecording {
     // max_capture_length specifies the maximum packet length.
     // Full packet length will be captured if absent/zero.
     uint32 max_capture_length = 3;
+
+    // stop_condition defines conditions which will cause the recording to
+    // stop early after any of the stop conditions has been hit
+    StopCondition stop_condition = 4;
+}
+
+// StopCondition defines one or more conditions which cause the recording to
+// stop after they have been hit. Stop conditions are ignored if they are
+// absent or zero-valued. If multiple conditions are defined, the recording
+// stops after the first one is hit.
+message StopCondition {
+    // bytes_captured stops the recording after at least this many bytes have
+    // been captured. Note: The resulting file might be slightly larger due
+    // to added pcap headers.
+    uint64 bytes_captured = 1;
+    // bytes_captured stops the recording after at least this many packets have
+    // been captured.
+    uint64 packets_captured = 2;
```

Comment:
nit: I think that `bytes_captured_count` and `packets_captured_count` would be better names because the current ones sound a bit like they'd be boolean values.

Reply:
also `num_bytes` and `num_captured` could work or even `bytes/packets` one word

Reply:
Sure, will change!

Reply:
Going with `packets/bytes_captured_count`. The reason I prefer to actually include `captured` is to make clear that it refers to these values here:

https://github.com/cilium/cilium/blob/6f9b87549f23effae024934ec3fcdec4dba95aa3/api/v1/recorder/recorder.proto#L110-L121

We might want to add other filters in the futures, e.g. `bytes_lost_count` or maybe even `bytes_written_count` (for the actual bytes written to disk, which includes things like pcap headers)